src/main/java/org/truffleruby/language/exceptions/TopLevelRaiseHandler.java

```diff
@@ -80,4 +88,13 @@ public void setLastException(VirtualFrame frame, DynamicObject exception) {
         setExceptionVariableNode.setLastException(frame, exception);
     }
 
+    public void handleSignalException(DynamicObject exception) {
+        if (Layouts.BASIC_OBJECT.getLogicalClass(exception) == coreLibrary().getSignalExceptionClass()) {
+            int signo = castToInt(ReadObjectFieldNodeGen.getUncached().execute(exception, "@signo", null));
+            DynamicObject posixModule = coreLibrary().getPosixModule();
+            CallDispatchHeadNode node = CallDispatchHeadNode.createPrivate();
```

Comment:
This node should be adopted and assigned to a `@Child` field, or `CallDispatchHeadNode#getUncached` should be used.

Reply:
Removed in later code.